custom_images/custom-datahub-actions/mirror_action.py

```diff
@@ -0,0 +1,222 @@
+# read-modify-write requires access to the DataHubGraph (RestEmitter is not enough)
+from datahub.ingestion.graph.client import DatahubClientConfig, DataHubGraph
+# Imports for metadata model classes
+from datahub.metadata.schema_classes import (
+    ChangeTypeClass,
+    GlobalTagsClass,
+    TagAssociationClass,
+    OwnerClass,
+    OwnershipClass,
+    EditableDatasetPropertiesClass,
+    EditableSchemaMetadataClass,
+    EditableSchemaFieldInfoClass,
+    AuditStampClass,
+)
+import re
+import os
+
+from gql_get_sb_regions import get_sb_regions
+
+ENV = os.getenv('ENV', 'stg')
+GMS_ENDPOINT = os.getenv('GMS_ENDPOINT', '')
+GMS_TOKEN = ''
+
+logger = logging.getLogger(__name__)
+
+graph = DataHubGraph(
+    DatahubClientConfig(
+        server=GMS_ENDPOINT,
+        token=GMS_TOKEN,
+    ))
+
+
+def replaceUrnFromSourceToTarget(urn: str,
+                                 target_sb_region: str,
+                                 sb_region_index_from_name: int,
+                                 is_elasticsearch: bool = False) -> str:
+  urn_splitted = urn.split(',')
+  table_name = urn_splitted[1]
+  splitted = table_name.split('.')
+  source_sb_region = splitted[sb_region_index_from_name]
+  splitted[sb_region_index_from_name] = target_sb_region
+  processed_table_name = '.'.join(splitted)
+  if is_elasticsearch:
+    processed_table_name = processed_table_name.replace(
+        source_sb_region, target_sb_region)
+  urn_splitted[1] = processed_table_name
+  return ','.join(urn_splitted)
+
+
+class MirrorAction(Action):
+  @classmethod
+  def create(cls, config_dict: dict, ctx: PipelineContext) -> "Action":
+    logger.info(ctx)
+    logger.info(config_dict)
+    gql_query = config_dict['gql_query']
+    gql_query = gql_query.replace('{env}', ENV.lower())
+    table_name_template = config_dict['table_name_template']
+    platform = config_dict['platform']
+    # For BigQuery, name is like `sb-dw-mesg-prod.log_access.ap1`. When this string is splitted by '.', the index that points to sb_region is -1.
+    # For others, name is like `ap1.soda.main_aametrics`. Thus, sb_region_index_from_name is 0.
+    sb_region_index_from_name = 0 if platform != 'bigquery' else -1
+    source_sb_region = config_dict[
+        'source_sb_region'] if ENV == 'prod' else config_dict[
+            'source_sb_region_stg']
+
+    cls.target_sb_regions = get_sb_regions(
+        gms_endpoint=GMS_ENDPOINT,
+        token=GMS_TOKEN,
+        gql_query=gql_query,
+        platform=platform,
+        sb_region_index_from_name=sb_region_index_from_name,
+    )
+    cls.platform = platform
+    cls.table_name_template = table_name_template
+    cls.sb_region_index_from_name = sb_region_index_from_name
+    cls.source_sb_region = source_sb_region
+
+    return cls(ctx)
+
+  def __init__(self, ctx: PipelineContext):
+    self.ctx = ctx
+    logger.info(self.target_sb_regions)
+
+  def act(self, event: EventEnvelope) -> None:
```

Comment:
This method is a quite big, and contains too many indentations. 
Why don't we split this method into a multiple, smaller methods? Also, consider the following coding style to reduce the indentation: 
```
# as-is 
if aspect: 
  do the task here

# alternative 
if not aspect: 
  return 

do the task here
```
This style, and spliting a method into multiple methods helps reducing a nested indentation.

Reply:
I apply your suggestion! I appreciate the suggestion. @sendbird-sehwankim